mountd/rpc_lsarpc.c

```diff
@@ -760,7 +760,9 @@ int rpc_lsarpc_init(void)
 	 * ksmbd supports the standalone server and
 	 * uses the hostname as the domain name.
 	 */
-	gethostname(domain_string, NAME_MAX);
+	if (gethostname(domain_string, NAME_MAX))
```

Comment:
we don't check return value of rpc_samr_init() and rpc_lsarpc_init(). I think that we need to add error handling for them.